### PR TITLE
Replicate page title and description in OpenGraph tags

### PR DESCRIPTION
**Pull request by zarino**
This way, users sharing inner pages of TWFY (eg: MP profiles) on
Facebook will benefit from the customised text that’s currently
hidden away in the `<meta type="description">` element.

Part of #1118.

### CODE REVIEW

**Comment by zarino** [comment]
This assumes `<meta name="description">` is populated for all the pages on the site. As it happens, it’s empty on the TWFY homepage!!

Maybe there should be a default fallback for `$meta_description` on line 90 of https://github.com/mysociety/theyworkforyou/blob/623ded18aedafac29150f1202d40df85045e0296/classes/Renderer/Header.php#L90 ?

File: www/includes/easyparliament/templates/html/header.php [hunk 1]
```diff
@@ -18,11 +18,12 @@
...
     <link rel="home" title="Home" href="http://<?= DOMAIN ?>/">
 
-    <meta property="og:title" content="TheyWorkForYou">
+    <meta property="og:site_name" content="TheyWorkForYou">
+    <meta property="og:title" content="<?= preg_replace('#<[^>]*>#', '', $page_title) ?>">
     <meta property="og:type" content="website">
     <meta property="og:url" content="http://<?= DOMAIN ?>">
     <meta property="og:image" content="http://<?= DOMAIN ?>/images/favicon-256.png">
-    <meta property="og:description" content="TheyWorkForYou is a website which makes it easy to keep track of your local MP's activities.">
+    <meta property="og:description" content="<?= _htmlentities($meta_description) ?>">
```
Comment (struan) [inline www/includes/easyparliament/templates/html/header.php]: Not all pages have a metadescription so we should fallback to the original if it's missing.

Comment (zarino) [inline www/includes/easyparliament/templates/html/header.php]: I have just pushed a commit that should fix that, by adding a non-empty fallback in the Header.php class. https://github.com/mysociety/theyworkforyou/pull/1121/commits/013cee1b815b5d1381c61a273ad4728080816638

**Comment by struan** [comment]
Looks good to me.

**Comment by zarino** [comment]
From IRC earlier:

> @zarino: I'm thinking the description could be much better on an MP's /votes page – mentioning some of the vote categories for example, and not bothering with the contributions/speeches/questions stuff…
> @zarino: …but I can't work out where to add it in the PHP controller for the /votes page
> @struan: I'll have a look tomorrow

After some more digging I think maybe it's best as an `if($pagetype == 'votes')` after the MP `$desc` is created from the MP’s hansard data in [www/docs/mp/index.php](https://github.com/mysociety/theyworkforyou/blob/master/www/docs/mp/index.php#L176-L251). Do you think that's sensible, @struan?

**Comment by struan** [comment]
@zarino: yup, that looks as good a place as any.

**Pull request merged**